### DISCUSSION TITLE
How to secure the create user endpoint?

### DISCUSSION THREAD

**Discussion by alvinsga** [pocketbase/pocketbase#2030] [Q&A]
I am worried that someone may spam create multiple accounts using the create user endpoint. I have not set API rules on this because it needs to be available to users that are not logged in.

How do you typically prevent malicious actors from abusing this endpoint?

**Comment by ganigeorgiev** [comment]
You can define a ratelimiter on reverse proxy level (nginx, apache, caddy, etc.).

Eventually something similar will be integrated in PocketBase, but for now remains a low priority (you can subscribe to https://github.com/pocketbase/pocketbase/issues/1232).

Adding a captcha for the create/update action in general is also in my todo to research it (https://github.com/pocketbase/pocketbase/issues/731).

Additionally, while the above will not stop users to create multiple accounts, you can restrict access to some of your other collections by requiring the account to be `verified` (`@request.auth.verified`).
